compendium/LogService/include/cppmicroservices/logservice/Logger.hpp

```diff
@@ -0,0 +1,343 @@
+#ifndef CPPMICROSERVICES_LOGGER_H_
+#define CPPMICROSERVICES_LOGGER_H_
+
+#ifdef AUDIT
+#undef AUDIT
+#endif
+#ifdef ERROR
+#undef ERROR
+#endif
+#ifdef WARN
+#undef WARN
+#endif
+#ifdef INFO
+#undef INFO
+#endif
+#ifdef DEBUG
+#undef DEBUG
+#endif
+#ifdef TRACE
+#undef TRACE
+#endif
```

Comment:
Looks like 'ERROR' has some conflicting macro defined in some Windows header, which is resulting in errors in our code if we don't undefine 'ERROR' . Rest of the #undef 's are not required, and I have removed them.

[Ideally, an enum class should scope its enumerators appropriately, preventing name conflicts. However, certain macros in the Windows headers can cause unexpected behavior when their names are reused, even within the scope of an enum class.]

Reply:
Ideally, its good practice to avoid use of ALL_CAPS for enumerators. See - https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum5-dont-use-all_caps-for-enumerators